### PR TITLE
14801 - Extract small HTML elements from NavigationTree to Twig

### PR DESCRIPTION
**Pull request by danielTiringer**
### Description

Extract HTML from `libraries/classes/Navigation/NavigationTree.php`'s small functions to Twig templates

Ref #14801

Before submitting pull request, please review the following checklist:

- [X] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [X] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [X] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [X] Every commit has a descriptive commit message.
- [X] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [X] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
💯

File: templates/navigation/tree/pagination_params.twig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% if position %}
+    <span
+        class="hide pos{{ position }}_nav"
```
Comment (mauriciofauth) [inline templates/navigation/tree/pagination_params.twig:3]: Do not break the class name.

File: libraries/classes/Navigation/NavigationTree.php [hunk 2]
```diff
@@ -1077,23 +1077,21 @@ public function renderPath()
...
-            $retval .= ' data-name="' . $paths['aPath_clean'][4] . '"';
-            $retval .= ' data-value="' . htmlspecialchars((string) $node->pos3) . '"';
-            $retval .= '"></span>';
+            $renderDetails['position'] = 3;
+            $renderDetails['data_name'] = $paths['aPath_clean'][4];
+            $renderDetails['data_value'] = (string) $node->pos3;
         }
 
-        return $retval;
+        return $this->template->render('/navigation/tree/pagination_params', $renderDetails);
```
Comment (mauriciofauth) [inline libraries/classes/Navigation/NavigationTree.php:1094]:
```suggestion
        return $this->template->render('navigation/tree/pagination_params', $renderDetails);
```

File: libraries/classes/Navigation/NavigationTree.php [hunk 3]
```diff
@@ -1671,20 +1666,15 @@ public static function sortNode(Node $a, Node $b): int
...
-            $retval .= RecentFavoriteTable::getInstance('favorite')
-                ->getHtml();
+            $renderDetails['favorite'] = RecentFavoriteTable::getInstance('favorite')->getHtml();
         }
 
-        $retval .= '<div class="clearfloat"></div>';
-        $retval .= '</div>';
-
-        return $retval;
+        return $this->template->render('/navigation/tree/quick_warp', $renderDetails);
```
Comment (mauriciofauth) [inline libraries/classes/Navigation/NavigationTree.php:1678]:
```suggestion
        return $this->template->render('navigation/tree/quick_warp', $renderDetails);
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**